adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java

```diff
@@ -303,6 +307,7 @@ public void run() {
         } else {
             Logger.v(TAG, "Reuse webview");
         }
+        Telemetry.getInstance().stopEvent(mAuthRequest.getTelemetryRequestId(), uiEvent, EventStrings.UI_EVENT);
```

Comment:
>Telemetry.getInstance().stopEvent(mAuthRequest.getTelemetryRequestId(), uiEvent, EventStrings.UI_EVENT); [](start = 8, length = 104)

I don't think you should have the stop event in onDestroy instead of here, onCreate will be followed by onStart, onResume, it doesn't mean the UI is already done.